roles/grafana_cloud_operator/tasks/delete_grafana_oncall_hub_spoke.yml

```diff
@@ -0,0 +1,156 @@
+
+- name: Print delete_integration_for
+  ansible.builtin.debug:
+    var: delete_integration_for
+
+- name: Extract integration details from the response
+  ansible.builtin.set_fact:
+    grafana_integrations: "{{ existing_integrations_response.json.results }}"
+
+- name: Find Grafana Integration IDs for the Integrations to Delete
+  ansible.builtin.set_fact:
+    integrations_to_delete: "{{ integrations_to_delete | default([]) + [{'name': item, 'integration_id': integration_id}] }}"
+  loop: "{{ delete_integration_for }}"
+  vars:
+    integration_id: >-
+      {{
+        grafana_integrations | selectattr('name', 'eq', item) | map(attribute='id') | first | default(None)
+      }}
+  when: integration_id is not none
+
+- name: Delete integrations from Grafana Cloud for deleted ManagedClusters
+  ansible.builtin.uri:
+    url: "{{ grafana_cloud_operator_grafana_cloud_integrations_api_url }}{{ item.integration_id }}"
+    method: DELETE
+    headers:
+      Authorization: "{{ grafana_cloud_api_token }}"
+      Content-Type: "application/json"
+    status_code: [200, 201]
+  loop: "{{ integrations_to_delete }}"
+  loop_control:
+    label: "{{ item }}"
+  retries: 5
+  delay: 6
+  when: delete_integration_for
+  failed_when: false
+
+- name: Update CR status for Integrations Deleted
+  operator_sdk.util.k8s_status:
+    api_version: grafanacloud.stakater.com/v1alpha1
+    kind: Config
+    name: "{{ cr_name }}"
+    namespace: "{{ cr_namespace }}"
+    status:
+      managedClusters: "{{ current_managed_clusters }}"
+      conditions:
+        - lastTransitionTime: "{{ ansible_date_time.iso8601 }}"
+          status: "True"
+          type: "DeletionCompleted"
+          reason: "OperationDone"
+          message: "Deleted the integrations {{ delete_integration_for }}"
+  when: delete_integration_for
+
+- name: Decode and modify alertmanager secret configuration
+  ansible.builtin.set_fact:
+    encoded_alertmanager_secret_content: "{{ {} | to_nice_yaml | b64encode }}"
+
+- name: Empty ManifestWork for each cluster to patch alertmanager-main secret
+  kubernetes.core.k8s:
+    state: present
+    definition:
+      apiVersion: work.open-cluster-management.io/v1
+      kind: ManifestWork
+      metadata:
+        name: "{{ item.name }}-manifestwork-grafana-oncall"
+        namespace: "{{ item.name }}"
+      spec:
+        workload:
+          manifests:
+            - apiVersion: v1
+              kind: Secret
+              metadata:
+                name: alertmanager-main
+                namespace: openshift-monitoring
+              data:
+                alertmanager.yaml: "{{ encoded_alertmanager_secret_content }}"
+  loop: "{{ integrations_to_delete }}"
+  loop_control:
+    label: "{{ item }}"
+  when:
+    - delete_integration_for | length > 0
+  register: manifestwork_deletion_results
+
+- name: Update Object with no data for each cluster to patch alertmanager-main secret
```

Comment:
What's the difference between this one and the above step? 🤔

Reply:
Above task empties the secret and this one removes the manifestwork cr

Reply:
won't it accomplish the same thing if we just delete the CR?

Reply:
Deleting CR doesn't remove the data from Spokes. It would remain as leftover